packages/flutter/test/material/dialog_test.dart

```diff
@@ -102,6 +102,53 @@ void main() {
     expect(materialWidget.type, MaterialType.card);
     expect(materialWidget.elevation, 24);
     expect(materialWidget.color, Colors.grey[800]);
+    expect(materialWidget.shape, const RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(2.0))));
+  });
+
+  testWidgets('Custom dialog shape', (WidgetTester tester) async {
```

Comment:
You should include a test that verifies that a null border DTRT as well.

Reply:
Done, I also abstracted out a helper method to reduce some of the boilerplate around the `AlertDialog`s, let me know if you prefer how it was before.